Helper/Cart.php

```diff
@@ -2318,8 +2319,21 @@ public function collectDiscounts(
 
                         break;
                 }
+                $amount -= $roundedAmount;
                 $totalAmount -= $roundedAmount;
             }
+            if ($this->deciderHelper->isAPIDrivenIntegrationEnabled()
+                && $this->deciderHelper->isSkipCartDiscountTotalMismatch()
+                && $amount) {
+                uasort($discounts, function ($a, $b) {
+                    return $b['amount'] <=> $a['amount'];
+                });
+                $firstKey = array_key_first($discounts);
+                $discounts[$firstKey]['amount'] += $amount;
+                ksort($discounts);
+                $totalAmount -= $amount;
+                $this->bugsnag->notifyError('Cart discount total mismatch', "Skip cart discount total mismatch. Actual Mismatch {$amount}. Discounts details: (before rounding)" . var_export($ruleDiscountDetails, true) . " (after rounding) " .var_export($discounts, true));
```

Comment:
Have we tested that $ruleDiscountDetails is not deeply nested object and var_export($ruleDiscountDetails, true) works well?

Reply:
$ruleDiscountDetails only contains sale rule id and its discount amount.